### DISCUSSION TITLE
Deprecated or not deprecated

### DISCUSSION THREAD

**Discussion by telecodani** [laravel/framework#33343] [General] [answered]
Hi all,

I'm on my way to upgrade my application to Laravel 5.8.* version but I saw into the upgrade guide, https://laravel.com/docs/5.8/upgrade#support, that helpers like array_* are now trated as deprecated. I'm confused because the new method, Arr, internally is using the array_* helpers. Are really deprecated? So, my question is, can I still using the helpers array_* as a long term solution for my application? I can't find a replace for array_search...

The change/pull on the framework:
https://github.com/laravel/framework/pull/26898

Laravel 5.8 - Arr
https://github.com/laravel/framework/blob/5.8/src/Illuminate/Support/Arr.php
Laravel 7.X- Arr - array_* still there
https://github.com/laravel/framework/blob/7.x/src/Illuminate/Support/Arr.php

**Comment by devcircus** [comment, marked as answer]
The array_* helpers that you are seeing in Laravel 7 are actual php functions. These are not Laravel helpers. You can still use the helpers that once existed in Laravel by pulling in laravel/helpers.

**Reply by devcircus** [reply]
See the [php docs](https://www.php.net/manual/en/ref.array.php) for the available array helpers in php, which includes ```array_search```.

**Reply by telecodani** [reply]
Understood. I have confused the Laravel helpers with php helpers. So, in this case, I just have to be aware of changes on new php versions that affect those array helpers. Thanks for your help.
